libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -772,7 +775,14 @@ namespace eosio { namespace chain { namespace backing_store {
          primary_key = key;
       }
 
-      return primary_iter_store.add(primary_key_iter(table_ei, *primary_key, found_payer));
+      const auto store_itr = primary_iter_store.add(primary_key_iter(table_ei, *primary_key, found_payer));
+
+      // value pointed by the iterator might have changed. need to cache from the      source.
```

Comment:
We shouldn't have to go through constructing the key again and retrieving the value, since session_iter points to the key and value already.

Should change line769 to:
payer_payload pp(*(session_iter).second);
const account_name found_payer = pp.payer:

and change lines 780-783 to be:
primary_iter_store.set_value(store_itr, pp.value, pp.value_size);